### PR TITLE
Rework of resource consumer.

### LINKED ISSUE #21664: Change HPA e2e tests to not rely on master proxy
Instead of generating load from inside of e2e there should be created a pod, which does it.

Forked from #20925

### PR DESCRIPTION
**Pull request by jszczepkowski**
Major rework of resource consumer: added controller running as a pod that spreads requests around consumers. This should fix #21664 and #23536.

### CODE REVIEW

File: test/e2e/autoscaling_utils.go [hunk 1]
```diff
@@ -40,7 +40,10 @@ const (
 	timeoutRC                       = 120 * time.Second
 	startServiceTimeout             = time.Minute
 	startServiceInterval            = 5 * time.Second
-	resourceConsumerImage           = "gcr.io/google_containers/resource_consumer:beta2"
+	// resourceConsumerImage           = "gcr.io/google_containers/resource_consumer:beta2"
```
Comment (jszczepkowski) [inline test/e2e/autoscaling_utils.go]: I'll create proper public images after we are done with the review.

**Comment by piosz** [comment]
```
Boilerplate header is wrong for: /go/src/k8s.io/kubernetes/test/images/resource-consumer/common/common.go
FAILED   ./hack/../hack/verify-boilerplate.sh	0s
```

**Comment by jszczepkowski** [comment]
@piosz 
> Boilerplate header is wrong for: /go/src/k8s.io/kubernetes/test/images/resource-consumer/common/common.go
FAILED   ./hack/../hack/verify-boilerplate.sh   0s

Fixed

File: test/images/resource-consumer/Makefile [hunk 2]
```diff
@@ -20,6 +20,7 @@ all: clean consumer
 consumer:
 	CGO_ENABLED=0 go build -a -installsuffix cgo --ldflags '-w' -o consume-cpu/consume-cpu ./consume-cpu/consume_cpu.go
 	CGO_ENABLED=0 go build -a -installsuffix cgo --ldflags '-w' -o consumer .
+	CGO_ENABLED=0 go build -a -installsuffix cgo --ldflags '-w' -o controller/controller ./controller/controller.go
 
 container: image
```
Comment (piosz) [inline test/images/resource-consumer/Makefile]: You may want to add here a command to build also the controller image.

**Comment by piosz** [comment]
LGTM, just a nit.
Please fix issue with image

File: test/e2e/autoscaling_utils.go (hunk 1 above)
Comment (jszczepkowski) [inline test/e2e/autoscaling_utils.go]: Done.

**Comment by jszczepkowski** [comment]
> Please fix issue with image

Done

File: hack/verify-flags/known-flags.txt [hunk 3]
```diff
@@ -463,3 +466,4 @@ watch-only
 whitelist-override-label
 windows-line-endings
 www-prefix
+
```
Comment (piosz) [inline hack/verify-flags/known-flags.txt]: empty line

**Pull request merged**